File: src/protocol/TemperatureDemandCommand.cpp
#include "TemperatureDemandCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

TemperatureDemandCommand::TemperatureDemandCommand()
        : Command(10, CommandType::TEMPERATURE_DEMAND)
{}

TemperatureDemandCommand::~TemperatureDemandCommand()
{}

std::vector<uint8_t> TemperatureDemandCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

string TemperatureDemandCommand::getName()
{
    return string("TemperatureDemandCommand");
}

void TemperatureDemandCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void TemperatureDemandCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/protocol/TemperatureDemandCommand.java
package protocol;

import java.util.ArrayList;

public final class TemperatureDemandCommand extends Command
	{
		public TemperatureDemandCommand()
		{
			super(10, CommandType.TEMPERATURE_DEMAND);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("TemperatureDemandCommand");
		}

		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}